pkg/cephfs/controllerserver.go

```
@@ -86,8 +84,11 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 
 	// Existence and conflict checks
-	idLk := volumeNameLocker.Lock(requestName)
-	defer volumeNameLocker.Unlock(idLk, requestName)
+	if acquired := cs.VolumeLocks.TryAcquire(requestName); !acquired {
```

Move this below line 80 (fail fast)